pvlib/tests/iotools/test_pvgis.py

```diff
@@ -435,6 +435,22 @@ def _compare_pvgis_tmy_basic(expected, meta_expected, pvgis_data):
         assert np.allclose(data[outvar], expected[outvar])
 
 
+@pytest.mark.remote_data
+@pytest.mark.flaky(reruns=RERUNS, reruns_delay=RERUNS_DELAY)
+def test_get_pvgis_tmy_coerce_year():
+    """test utc_offset and coerce_year work as expected"""
+    pvgis_data = get_pvgis_tmy(45, 8, utc_offset=2)  # Turin
+    jan1_midnight1990 = pd.Timestamp('1990-01-01 00:00:00', tz='Etc/GMT-2')
+    dec31_midnight1990 = pd.Timestamp('1990-12-31 23:00:00', tz='Etc/GMT-2')
+    assert pvgis_data[0].index[0] == jan1_midnight1990
+    assert pvgis_data[0].index[-1] == dec31_midnight1990
+    pvgis_data = get_pvgis_tmy(45, 8, utc_offset=2, coerce_year=2021)  # Turin
+    jan1_midnight2021 = pd.Timestamp('2021-01-01 00:00:00', tz='Etc/GMT-2')
+    dec31_midnight2021 = pd.Timestamp('2021-12-31 23:00:00', tz='Etc/GMT-2')
+    assert pvgis_data[0].index[0] == jan1_midnight2021
+    assert pvgis_data[0].index[-1] == dec31_midnight2021
```

Comment:
Can we also check that a few values are aligned with the index correctly?

Reply:
I checked noon for every month with default and coerced year. Note I conflated a few other updates to the test:
- check that UTC time zone is unchanged if using defaults
- update Turin is actually CET which is UTC+1, not +2, oops! I got confused because currently it is CEST, or summertime which _is_ +2, 😆 ha-ha
- use some variables to be DRY to replace some WET & hardcoded args like the coerced year used 3x, time zone used 4x, and the "Etc/GMT" tz name also used 4x
- I didn't test the combo of `utc_offset` is zero, and `coerce_year` is set - should do nothing - do we need this test?